lib/facter/resolvers/jail.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    class Jail < BaseResolver
+      init_resolver
+
+      class << self
+        private
+
+        def post_resolve(fact_name)
+          @fact_list.fetch(fact_name) { check_jail(fact_name) }
+        end
+
+        def check_jail(fact_name)
+          @fact_list[:vm] = is_jailed
+          @fact_list[fact_name]
+        end
+
+        def is_jailed
+          require_relative 'freebsd/ffi/ffi_helper'
+          if Facter::Freebsd::FfiHelper.sysctl_by_name(:long, 'security.jail.jailed').zero?
```

Comment:
This code does not make sense outside of FreeBSD.  How would you recommend to manage this?